### PR TITLE
8340605: Open source several AWT PopupMenu tests

### PR DESCRIPTION
**Pull request by kumarabhi006**
Several AWT PopupMEnu related tests are converted from applet to manual and moved to open.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8340605](https://bugs.openjdk.org/browse/JDK-8340605): Open source several AWT PopupMenu tests (**Bug** - P4)

### Reviewers
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21151/head:pull/21151` \
`$ git checkout pull/21151`

Update a local copy of the PR: \
`$ git checkout pull/21151` \
`$ git pull https://git.openjdk.org/jdk.git pull/21151/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21151`

View PR using the GUI difftool: \
`$ git pr show -t 21151`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21151.diff">https://git.openjdk.org/jdk/pull/21151.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21151#issuecomment-2370462220)

### CODE REVIEW

File: test/jdk/java/awt/PopupMenu/PopupLeadingSeparatorTest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * @run main/manual PopupLeadingSeparatorTest
+ */
+
+public class PopupLeadingSeparatorTest {
+    public static void main(String[] args) throws Exception {
+        PopupLeadingSeparatorTest obj = new PopupLeadingSeparatorTest();
+        String INSTRUCTIONS = """
+                Press mouse button on the frame. Popup menu without leading
+                separator should appear.
+                If a PopupMenu behaves same, press Pass, else press Fail.
```
Comment (TejeshR13) [inline test/jdk/java/awt/PopupMenu/PopupLeadingSeparatorTest.java:48]:
```suggestion
                If a PopupMenu behaves same press Pass, else press Fail.
```

File: test/jdk/java/awt/PopupMenu/PopupMenuWithMenuBar.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ * @test
+ * @bug 4038140
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Test for functionality of PopupMenuWithMenuBar
+ * @run main/manual PopupMenuWithMenuBar
+ */
+
+public class PopupMenuWithMenuBar {
+
```
Comment (TejeshR13) [inline test/jdk/java/awt/PopupMenu/PopupMenuWithMenuBar.java:43]:
```suggestion
```

File: test/jdk/java/awt/PopupMenu/PopupOnButton.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ * @run main/manual PopupOnButton
+ */
+
+public class PopupOnButton {
+   public static void main(String[] args) throws Exception {
+       PopupOnButton obj = new PopupOnButton();
+       String INSTRUCTIONS = """
+                Right-click on the button.
+                Popup Menu should appear and behave fine.
+                If a PopupMenu appears, press Pass, else press Fail.
```
Comment (TejeshR13) [inline test/jdk/java/awt/PopupMenu/PopupOnButton.java:48]:
```suggestion
                If a PopupMenu appears press Pass, else press Fail.
```

File: test/jdk/java/awt/PopupMenu/PopupLeadingSeparatorTest.java (hunk 1 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/PopupMenu/PopupLeadingSeparatorTest.java:48]: seems correct to me.

File: test/jdk/java/awt/PopupMenu/PopupOnButton.java (hunk 3 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/PopupMenu/PopupOnButton.java:48]: same here.

**Review by TejeshR13 — APPROVED** [review]
LGTM.

**Pull request closed without merging**